### PR TITLE
Hack app draft

### PR DESCRIPTION
**Pull request by hassanabidpk**
Kindly check the model and suggest changes.

### CODE REVIEW

**Comment by jinpark** [comment]
Did a quick read through and this looks really good.

File: hacks/models.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    writer = models.ForeignKey(User)
+    updatedAt =  models.DateTimeField(auto_now=True)
+    createdAt =  models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return self.title
+
+    def save(self):
+        if not self.id:
+            self.createdAt = timezone.now()
```
Comment (jinpark) [inline hacks/models.py]: I don't think you need to do this since we have `auto_now` in the field

File: hacks/models.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            self.post_slug = slugify(self.title)
```
Comment (jinpark) [inline hacks/models.py]: Should this be `slug` instead of `post_slug`

File: hacks/models.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        self.updatedAt = timezone.now()
```
Comment (jinpark) [inline hacks/models.py]: Same comment as above, not sure if we need this if we have `auto_now_add`

File: hacks/models.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+class Category(models.Model):
+    title = models.CharField(max_length=200)
+    slug = models.SlugField(max_length=60, unique=True)
+    description = models.TextField()
+
+    class Meta:
+        verbose_name_plural = "Categories"
+
+    def __unicode__(self):
```
Comment (jinpark) [inline hacks/models.py]: If we're using python 3, shouldnt this be `__str__`

if we're not the hack model should use `__unicode__`

File: hacks/admin.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+from django.contrib import admin
+from .models import Hack
+
+admin.site.register(Hack)
```
Comment (jinpark) [inline hacks/admin.py]: should we add `Category` here as well?

File: hacks/tests.py [hunk 6]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (jinpark) [inline hacks/tests.py]: @hassanabidpk do you want to try writing some tests? I can help you out if needed

File: hacks/models.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        return self.title
+
+    def save(self):
+        if not self.id:
+            self.createdAt = timezone.now()
+            self.post_slug = slugify(self.title)
+        self.updatedAt = timezone.now()
+        return super(Hack,self).save()
+
+class CategoryToHack(models.Model):
```
Comment (jinpark) [inline hacks/models.py]: Not sure if we need this model. I think django works OOTB without referencing a join table

**Comment by jinpark** [comment]
Theres also a merge conflict. Probably just need to merge master into the branch

File: hacks/models.py (hunk 1 above)
Comment (hassanabidpk) [inline hacks/models.py]: ok

File: hacks/models.py (hunk 2 above)
Comment (hassanabidpk) [inline hacks/models.py]: agree

File: hacks/tests.py (hunk 6 above)
Comment (hassanabidpk) [inline hacks/tests.py]: okay sounds good

File: hacks/models.py (hunk 7 above)
Comment (hassanabidpk) [inline hacks/models.py]: good one

File: hacks/tests.py (hunk 6 above)
Comment (hassanabidpk) [inline hacks/tests.py]: sounds good

File: hacks/models.py (hunk 2 above)
Comment (hassanabidpk) [inline hacks/models.py]: got it-  (I think i messed up with branches on local computer) 2 branches of same name :p

File: hacks/models.py (hunk 3 above)
Comment (hassanabidpk) [inline hacks/models.py]: yeah right

**Pull request merged**